src/metabase/middleware/security.clj

```diff
@@ -29,46 +29,50 @@
-      (format "%s %s; " (name k) (apply str (interpose " " vs)))))})
+    (for [[k vs] {:default-src  ["'none'"]
+                  :script-src   ["'self'"
+                                 "'unsafe-eval'" ; TODO - we keep working towards removing this entirely
+                                 "https://maps.google.com"
+                                 "https://apis.google.com"
+                                 "https://www.google-analytics.com" ; Safari requires the protocol
+                                 "https://*.googleapis.com"
+                                 "*.gstatic.com"
+                                 ;; for webpack hot reloading
+                                 (when config/is-dev?
+                                   "localhost:8080")
+                                 ;; inline script in index.html that sets `MetabaseBootstrap` and the like
+                                 "'sha256-dPkijyoRrGWwHu+PrDIMt/5wQPFfVuvhU7vendgcUSQ='"
+                                 ;; Web Font Loader font configuration (WebFontConfig) in index.html
+                                 "'sha256-6xC9z5Dcryu9jbxUZkBJ5yUmSofhJjt7Mbnp/ijPkFs='"
+                                 ;; inline script in index.html that loads Google Analytics
+                                 "'sha256-uKEj/Qp9AmQA2Xv83bZX9mNVV2VWZteZjIsVNVzLkA0='"]
+                  :child-src    ["'self'"
+                                 ;; TODO - double check that we actually need this for Google Auth
+                                 "https://accounts.google.com"]
+                  :style-src    ["'self'"
+                                 "'unsafe-inline'" ; needed for Google Fonts
+                                 "fonts.googleapis.com"]
+                  :font-src     ["'self'"
+                                 "fonts.gstatic.com"
+                                 "themes.googleusercontent.com"
+                                 (when config/is-dev?
+                                   "localhost:8080")]
+                  :img-src      ["*"
+                                 "'self' data:"]
+                  :connect-src  ["'self'"
+                                 ;; MailChimp. So people can sign up for the Metabase mailing list in the sign up process
+                                 "metabase.us10.list-manage.com"
+                                 (when config/is-dev?
+                                   "localhost:8080 ws://localhost:8080")]
+                  :manifest-src ["'self'"]}]
+      (format "%s %s; " (name k) (str/join " " vs))))})
 
 (defsetting ssl-certificate-public-key
   (str (tru "Base-64 encoded public key for this site's SSL certificate.")
        (tru "Specify this to enable HTTP Public Key Pinning.")
        (tru "See {0} for more information." "http://mzl.la/1EnfqBf")))
 ;; TODO - it would be nice if we could make this a proper link in the UI; consider enabling markdown parsing
 
-#_(defn- public-key-pins-header []
```

Comment:
removed since this has been commented out and wasting space for like 4 years now. And as @flamber pointed out in #9639 it's a [dead fish now](https://www.theregister.co.uk/2017/10/30/google_hpkp/)